src/service/history-impl.js

```diff
@@ -109,6 +112,20 @@ export class History {
   }
 
   /**
+   *
+   * @param {string} target
+   * @param {string} previousTarget
```

Comment:
Not super happy about `previousTarget`. First, it's previousHash since it may not be a target. Second, seems like we should know it here anyway.